the-collab-lab/tcl-15-smart-shopping-list
# issue#10 | Estimate user's next purchase

## Description

This PR adds two new attributes to the item object, the `numberOfPurchases`, and the `previousPurchase`, It estimates how soon the user is going to buy the items and stores the dates as integers instead of strings.

## Related Issue

Closes #10

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull origin yk-se-issue10` and checkout that branch with `git checkout yk-se-issue10 ` 
* Run `npm start` to launch the app
* In the app, create a new list and add a few items
* Go to the List view and check one of the newly added list items. 
* Open up the [Firestore console](https://console.firebase.google.com/u/1/project/tcl-15-smart-shopping-list/firestore/data~2FshoppingLists~2FZc4ke3K09gJkaWmXE7uC) and take a look at the checked item, you should notice that previousPurchase and recentPurchase for the unchecked items are initialized to `null` and numberOfPuchases is 0. However,  the recentPurchase for the checked item has a value and the numberOfPurchases is 1.
* Now add a value less than the recent purchase to simulate that more than 24hrs has passed since the last purchase.
* Return to the app, you will notice the item is now unchecked.
* Check it a second time, return to the console and observe the value of `howSoon` update to reflect the estimated days till next purchase and the numberOfPurchases increase by 1.
